### ISSUE TITLE
Simple Django project deploys successfully on v218, fails on v219 and following

### ISSUE THREAD

**Issue by ehmatthes** [heroku/heroku-buildpack-python#1374]
Hi, I have a [simple Django project](https://github.com/ehmatthes/learning_log_2e_django40) that I've used to teach deployment on Heroku for years. Deployments have stopped working recently, and I believe it's related to the [new approach](https://github.com/heroku/heroku-buildpack-python/pull/1363) to deciding whether to auto-provision a Heroku Postgres database.

Here's an [issue](https://github.com/ehmatthes/learning_log_2e_django40/issues/1) detailing everything I've tried in an attempt to troubleshoot deployment. I am not an expert in Heroku buildpacks, but I did make a [demo buildpack](https://github.com/ehmatthes/heroku-buildpack-python/tree/my_autoconfigure_101721) a couple years ago that automated Django deployments for simple projects, so I have some idea how they work.

This project meets all three listed requirements for having a Heroku Postgres database auto-provisioned: it has a manage.py file in the root directory, and both `django` and `psycopg2` are listed in requirements.txt. I forked the current version of this buildpack, and added some [diagnostics](https://github.com/ehmatthes/heroku-buildpack-python/blob/main/bin/compile#L275) in `bin/compile` after `collectstatic` is run, which shows that the project does pass the three tests as written in `bin/release`.

The project deploys with the new buildpack, but it does not have a Postgres database provisioned. All pages that don't require a database work; pages that require a database fail. I have deployed to free dynos and to paid dynos, without seeing any change in behavior. 

- Is there any obvious reason this project fails to have a Postgres database auto-provisioned despite seeming to meet the three requirements?
- Where is the `bin/release` script called from? I tried to move my diagnostics closer to the release phase, but couldn't find where the release script is called, so placing it after `collectstatic` was the best I could do.

Thanks.

**Comment by ehmatthes** [comment]
Output of diagnostics:

```
remote: -----> $ python manage.py collectstatic --noinput
remote:        128 static files copied to '/tmp/build_43f0844b/staticfiles', 378 post-processed.
remote: 
remote: 
remote: *****> Begin diagnostic work...
remote: *****> BUILD_DIR: /tmp/build_43f0844b
remote: *****> BUILD_DIR contents:
remote: total 64K
remote: drwx--x--- 9 u58777 dyno 4.0K Oct  3 20:12 .
remote: drwx------ 5 u58777 dyno 4.0K Oct  3 20:12 ..
remote: -rw------- 1 u58777 dyno   42 Oct  3 20:12 .gitignore
remote: drwx------ 4 u58777 dyno 4.0K Oct  3 20:12 .heroku
remote: drwx------ 3 u58777 dyno 4.0K Oct  3 20:12 learning_log
remote: drwx------ 5 u58777 dyno 4.0K Oct  3 20:12 learning_logs
remote: -rw------- 1 u58777 dyno  623 Oct  3 20:12 manage.py
remote: -rw------- 1 u58777 dyno   44 Oct  3 20:12 Procfile
remote: drwx------ 2 u58777 dyno 4.0K Oct  3 20:12 .profile.d
remote: -rw------- 1 u58777 dyno  293 Oct  3 20:12 requirements.txt
remote: -rw------- 1 u58777 dyno   13 Oct  3 20:12 runtime.txt
remote: drwx------ 3 u58777 dyno 4.0K Oct  3 20:12 staticfiles
remote: drwx------ 2 u58777 dyno 4.0K Oct  3 20:12 templates
remote: -rw------- 1 u58777 dyno 5.0K Oct  3 20:12 test_deployed_app_functionality.py
remote: drwx------ 5 u58777 dyno 4.0K Oct  3 20:12 users
remote: *****> Is manage.py present?
remote:   Found manage.py.
remote: *****> Is django available?
remote:   Django is available.
remote: *****> Is psycopg2 available?
remote:   Django is available.
remote: *****> Are all three conditions met?
remote:   All three conditions met.
remote: *****> End diagnostic work.
remote: 
remote: -----> Discovering process types
remote:        Procfile declares types -> web
```

**Comment by edmorley** [comment]
@ehmatthes Thank you for the great report + testcase. I've tracked down the issue and have a fix in #1375 :-)
